pkgs/development/tools/jdt-language-server/default.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, jdk
+}:
+
+stdenv.mkDerivation rec {
+  pname = "jdt-language-server";
+  version = "0.67.0";
+  timestamp = "202012170459";
+
+  src = fetchurl {
+    url = "https://download.eclipse.org/jdtls/milestones/${version}/jdt-language-server-${version}-${timestamp}.tar.gz";
+    sha256 = "11qzwf2l737hdv1g7bfcrmdxcrl7g9z2k9nhy9lknlabmhq4mcdl";
+  };
+
+  sourceRoot = ".";
+
+  buildInputs = [
+    jdk
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    # Copy jars
+    install -D -t $out/share/java/plugins/ plugins/*.jar
+
+    # Copy config directories for linux and mac
+    mkdir -p $out/share/config_{linux,mac}/
+    install -D config_linux/* $out/share/config_linux/
+    install -D config_mac/* $out/share/config_mac/
+
+    # Get latest version of launcher jar
+    # e.g. org.eclipse.equinox.launcher_1.5.800.v20200727-1323.jar
+    launcher="$(ls $out/share/java/plugins/org.eclipse.equinox.launcher_* | sort -V | tail -n1)"
+
+    # jdt-language-server needs to write to the config location, so we create a temp
+    # directory each time and copy the config_{linux,mac}/ folders there
+    runtime_dir=''${XDG_CACHE_HOME:-$HOME/.cache}/jdt-language-server
+
+    makeWrapper ${jdk}/bin/java $out/bin/jdt-language-server \
```

Comment:
Does the language server need to be running with the same version the user is coding in ?
Gradle and Maven try using JAVA_HOME if it's set otherwise use the default JDK in the derivation.

Reply:
Yeah so as commented above, it needs to be the same version or higher. I don't know if there is a benefit to the user being able to specify the version used here. For now it's fixed to `jdk15`. What do you think?